modules/internal/ChapelArray.chpl

```diff
@@ -2795,6 +2795,19 @@ module ChapelArray {
       }
     }
 
+    /* Return the last element in the array. The array must be a
+       rectangular 1-D array.
+     */
+    proc back() {
+      if !isRectangularArr(this) || this.rank != 1 then
+        compilerError("back() is only supported on 1D rectangular arrays");
+
+      if boundsChecking && isEmpty() then
+        halt("back called on an empty array");
```

Comment:
The `halt` on really should be `throw`.

Reply:
PR #10191 just added halt wrappers for this case, so it should call one of them.  Eventually we may want it to be a `throw`, but not quite yet.